File: Sorting/Quick Sort/quickSort.ts
function quickSorter(arr: number[]): number[] {
    if (arr.length <= 1) {
        return arr;
    }

    const pivot = arr[Math.floor(arr.length / 2)];
    const left: number[] = [];
    const middle: number[] = [];
    const right: number[] = [];

    for (const element of arr) {
        if (element < pivot) {
            left.push(element);
        } else if (element === pivot) {
            middle.push(element);
        } else {
            right.push(element);
        }
    }

    return [...quickSorter(left), ...middle, ...quickSort(right)];
}

// Example usage:
const unsortArray: number[] = [3, 6, 8, 10, 1, 2, 1];
const sortArray: number[] = quickSorter(unsortedArray);
console.log(sortedArray);

File: Sorting/Quick Sort/quicksort.py
def quick_sort(arr):
    if len(arr) <= 1:
        return arr
    
    pivot = arr[len(arr) // 2]  
    left = [x for x in arr if x < pivot] 
    middle = [x for x in arr if x == pivot] 
    right = [x for x in arr if x > pivot] 

    left = quick_sort(left)
    right = quick_sort(right)
    
 
    return left + middle + right

if __name__ == "__main__":
    unsorted_list = [3, 6, 8, 10, 1, 2, 1]
    sorted_list = quick_sort(unsorted_list)
    print(sorted_list)